packages/playwright/src/matchers/toMatchSnapshot.ts

```diff
@@ -365,7 +369,8 @@ export async function toHaveScreenshot(
   optOptions: ToHaveScreenshotOptions = {}
 ): Promise<MatcherResult<NameOrSegments | { name?: NameOrSegments }, string>> {
   const testInfo = currentTestInfo();
-  if (!testInfo)
+  const step = currentStep();
```

Comment:
ditto, this whole file basically. You are solving your problem on the testInfo layer. It should "just work" for this matcher same way it should work for user steps and matchers.